src/surface_tracker/orientation.py

```diff
@@ -0,0 +1,97 @@
+        mapping = self.__mapping_default_to_rotated()
+        return self.__map_surface_corners(corner_dict, mapping)
+
+    def map_surface_corners_to_default(self, corner_dict):
+        mapping = self.__mapping_rotated_to_default()
+        return self.__map_surface_corners(corner_dict, mapping)
+
+    def __mapping_default_to_rotated(self):
+        default_orientation = CornerId.all_corners()
+        rotated_orientation = CornerId.all_corners(
+            starting_with=self.surface_start_with, clockwise=self.clockwise
+        )
+        return dict(zip(default_orientation, rotated_orientation))
+
+    def __mapping_rotated_to_default(self):
+        default_orientation = CornerId.all_corners()
+        rotated_orientation = CornerId.all_corners(
+            starting_with=self.surface_start_with, clockwise=self.clockwise
+        )
+        return dict(zip(rotated_orientation, default_orientation))
+
+    def __map_surface_corners(self, corner_dict, mapping):
+        updated_corner_dict = {}
+        for corner_name, coords in corner_dict.items():
+            updated_corner_dict[mapping[corner_name]] = coords
+        return updated_corner_dict
+
+    def get_relative_rotation(self, clockwise=True):
+        default_start_with = CornerId.all_corners()[0]
+        rotated_orientation = CornerId.all_corners(
+            starting_with=self.surface_start_with, clockwise=self.clockwise
+        )
+        clockwise_rotation = rotated_orientation.index(default_start_with) * 90
+        if not clockwise:
+            return -clockwise_rotation % 360
+        return clockwise_rotation
+
+    def get_visual_anchor_surface_space(self):
+        if self.surface_start_with == CornerId.TOP_LEFT and self.clockwise:
+            top_indicator_corners = [[0.3, 0.7], [0.7, 0.7], [0.5, 0.9]]
+            top_indicator_corners.append(top_indicator_corners[0])
+        if self.surface_start_with == CornerId.TOP_RIGHT and self.clockwise:
+            top_indicator_corners = [[0.7, 0.7], [0.7, 0.3], [0.9, 0.5]]
+            top_indicator_corners.append(top_indicator_corners[0])
+        if self.surface_start_with == CornerId.BOTTOM_RIGHT and self.clockwise:
+            top_indicator_corners = [[0.7, 0.3], [0.3, 0.3], [0.5, 0.1]]
+            top_indicator_corners.append(top_indicator_corners[0])
+        if self.surface_start_with == CornerId.BOTTOM_LEFT and self.clockwise:
+            top_indicator_corners = [[0.3, 0.3], [0.3, 0.7], [0.1, 0.5]]
+            top_indicator_corners.append(top_indicator_corners[0])
+        return top_indicator_corners
+
+    def as_dict(self) -> dict:
+        return {
+            "surface_start_with": self.surface_start_with.name,
+            "clockwise": self.clockwise,
+        }
+
+    @staticmethod
+    def from_dict(value: dict) -> "SurfaceOrientation":
+        try:
+            return SurfaceOrientation(
+                surface_start_with=getattr(CornerId, value["surface_start_with"]),
```

Comment:
I think you can add a `@staticmethod` factory method to `CornerId` called `from_name(name: str)` (to be symmetric to the `name` property) that returns a `CornerId` instance if available, and use it here. It seems like there might be other places where this would be useful, and adding it would avoid code duplication. The implementation can be the same as here, it's just about abstraction.